### PR TITLE
Initial implementation of Red Hat Subscriptions checks

### PR DESCRIPTION
**Pull request by mirzal**
Few tier1 tests are ported already, but this is mostly to check implementation of [Airgun PR 163](https://github.com/SatelliteQE/airgun/pull/163).

Two tests require manual setup before they can be successfully run. This is caused by Robottelo fake manifest not supporting use-cases tested by them. I think this problem is bigger than this PR. AFAICT Jenkins won't run unmarked tests, so it won't interfere with job results.

Test results:

```
$ pytest -v tests/foreman/ui_airgun/test_subscription.py
test_subscription.py::test_positive_upload_and_delete PASSED         [ 10%]
test_subscription.py::test_negative_delete PASSED                    [ 20%]
test_subscription.py::test_positive_delete_confirmation PASSED       [ 30%]
test_subscription.py::test_read_provided_products PASSED             [ 40%]
test_subscription.py::test_read_enabled_products PASSED              [ 50%]
test_subscription.py::test_negative_read_enabled_products PASSED     [ 60%]
test_subscription.py::test_add_subscription PASSED                   [ 70%]
test_subscription.py::test_search_subscription PASSED                [ 80%]
test_subscription.py::test_negative_search_subscription PASSED       [ 90%]
test_subscription.py::test_remove_subscription PASSED                [100%]
```

### CODE REVIEW

File: tests/foreman/ui_airgun/test_subscription.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    with manifests.clone() as manifest:
+        with open(manifest.filename, 'wb') as handler:
+            handler.write(manifest.content.read())
+        return manifest.filename
+
+
+@skip_if_not_set('fake_manifest')
+@tier1
+@upgrade
+def test_positive_upload_and_delete(session, manifest_file, organization):
```
Comment (ldjebran) [inline tests/foreman/ui_airgun/test_subscription.py]: we need run_in_one_thread decorator

File: tests/foreman/ui_airgun/test_subscription.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        'Delete all subscriptions attached to activation keys.',
+        'Disable Red Hat Insights',
+        'Require you to upload the subscription-manifest and re-attach '
+        'subscriptions to hosts and activation keys.',
+        'This action should only be taken in extreme circumstances or for '
+        'debugging purposes.',
+    ]
+    with session:
+        session.organization.select(organization.name)
+        session.subscription.add_manifest(manifest_file)
```
Comment (ldjebran) [inline tests/foreman/ui_airgun/test_subscription.py]: think we do not need to add manifest via UI in this test, we can use API
we can use upload_manifest_locked function from manifests modules, in other case we have to decorate this function with run_in_one_thread.

**Comment by ldjebran** [comment]
The simplest solution is to mark this module to run in one thread with:

pytestmark = [run_in_one_thread]

**Comment by oshtaier** [comment]
closed according real good idea in https://github.com/SatelliteQE/airgun/pull/163#issuecomment-429341125

**Pull request closed without merging**